dashboard/src/app/dashboard/[dashboardId]/(dashboard)/WeeklyHeatmapSection.tsx

```diff
@@ -0,0 +1,165 @@
+    bounce_rate: 'Bounce Rate',
+    pages_per_session: 'Pages / Session',
+    session_duration: 'Session Duration',
+  }[selectedMetric];
+
+  const onMetricChange = (next: string) => {
+    setSelectedMetric(next as HeatmapMetric);
+  };
+
+  return (
+    <div className='bg-card border-border rounded-lg border p-6 shadow'>
+      <div className='mb-8 flex items-center justify-between'>
+        <div>
+          <h2 className='text-foreground mb-1 text-lg font-bold'>Weekly Trends</h2>
+        </div>
+        <div className='w-36'>
+          <Select value={selectedMetric} onValueChange={onMetricChange}>
+            <SelectTrigger className='w-full'>
+              <SelectValue />
+            </SelectTrigger>
+            <SelectContent>
+              {metricOptions.map((opt) => (
+                <SelectItem key={opt.value} value={opt.value}>
+                  {
+                    {
+                      pageviews: 'Pageviews',
+                      unique_visitors: 'Unique Visitors',
+                      sessions: 'Sessions',
+                      bounce_rate: 'Bounce Rate',
+                      pages_per_session: 'Pages / Session',
+                      session_duration: 'Session Duration',
+                    }[opt.value]
+                  }
+                </SelectItem>
+              ))}
+            </SelectContent>
+          </Select>
+        </div>
+      </div>
+
+      <HeatmapGrid
+        data={current?.matrix ?? []}
+        maxValue={current?.maxValue ?? 1}
+        metricLabel={selectedMetricLabel}
+        metric={selectedMetric}
+      />
+    </div>
+  );
+}
+
+type HeatmapGridProps = {
+  data: WeeklyHeatmapMatrix[];
+  maxValue: number;
+  metricLabel: string;
+  metric: HeatmapMetric;
+};
+
+function HeatmapGrid({ data, maxValue, metricLabel, metric }: HeatmapGridProps) {
+  const dayLabels = ['Mon.', 'Tue.', 'Wed.', 'Thu.', 'Fri.', 'Sat.', 'Sun.'];
+
+  const effectiveMax = Math.max(1, maxValue);
+
+  const getCellStyle = useCallback(
+    (value: number): CSSProperties => {
+      if (value <= 0) return {};
+
+      const t = Math.log1p(value) / Math.log1p(effectiveMax);
+      const eased = Math.pow(t, 0.85);
+
+      return {
+        backgroundColor: `oklch(62% 0.17 268.71 / ${0.1 + 0.9 * eased})`,
+      };
+    },
+    [effectiveMax],
+  );
+
+  return (
+    <div className='grid grid-cols-[40px_repeat(7,1fr)] gap-x-0.5 gap-y-1'>
+      <div></div>
+      {dayLabels.map((label) => (
+        <div
+          key={label}
+          className='text-muted-foreground truncate pb-1 text-center text-[10px] leading-none font-medium'
+        >
+          {label}
+        </div>
+      ))}
+
+      {Array.from({ length: 24 }).map((_, hourIndex) => (
+        <Fragment key={`hour-${hourIndex}`}>
+          <div className='text-muted-foreground flex h-2.5 items-center justify-end pr-2 text-xs leading-none'>
+            {hourIndex % 3 === 1 ? String(hourIndex).padStart(2, '0') : ''}
```

Comment:
It's a little confusion that it just says "01  04  07", as it could also represent a date, maybe add "01:00  04:00  07:00"? Else just resolve this 😎 

```suggestion
            {hourIndex % 3 === 1 ? `${String(hourIndex).padStart(2, '0')}:00` : ''}
```

I tried it out, and it looks like this:

<img width="748" height="488" alt="image" src="https://github.com/user-attachments/assets/46fae14b-e0a0-439f-8f2b-de534d7fe05a" />